pkg/registry/servicecatalog/broker/strategy.go

```diff
@@ -0,0 +1,99 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package broker
```

Comment:
I split the single `Strategy` object that I usually see into it's three components for clarity and a sense of understanding.

Not sure if there is a problem with referencing the `kapi.Scheme`. We may need to write our own.

Each of the Strategy's has an 'ObjectTyper' So we need one as well. The impl that fulfills it is `kapi.Scheme`.

There is a couple of method overlap between Create & Update, but at the toplevel they do not share an interface. I don't know what happens if they are different, or a reason why you would want a different canonicalize/namespacescoped between create and update of the same object.